gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js

```diff
@@ -413,11 +413,35 @@ gpii.windows.spi.setImpl = function (payload) {
     // to reload the default theme (or last saved?) - causing any custom wallpaper to be lost.
     // This hack just gets the current high-contrast state and does nothing if it's already turned off.
     var applyPromise;
-    if (payload.settings.HighContrastOn && !payload.settings.HighContrastOn.value &&
-        !payload.settings.HighContrastTheme) {
-        var currentState = gpii.windows.spi.getImpl(fluid.copy(payload));
-        if (!currentState.HighContrastOn.value) {
+    if (payload.settings.HighContrastOn) {
+        var doNothing = false;
+        if (!payload.settings.HighContrastOn.value && !payload.settings.HighContrastTheme) {
+            var currentState = gpii.windows.spi.getImpl(fluid.copy(payload));
+            if (!currentState.HighContrastOn.value) {
+                doNothing = true;
+            }
+        }
+
+        if (doNothing) {
             applyPromise = fluid.promise().resolve();
+        } else {
+            var cachedFilesPathEnv = "%APPDATA%\\Microsoft\\Windows\\Themes\\CachedFiles";
+            var cachedFilesPath = cachedFilesPathEnv.replace(/%([^%]+)%/g, function (_,n) { return process.env[n] });
+            var optionsCached = {
+                "recursive": true,
+                "silent": true,
+                "local": true
+            };
+
+            var tWallpaperEnv = "%APPDATA%\\Microsoft\\Windows\\Themes\\TranscodedWallpaper";
+            var tWallpaper = tWallpaperEnv.replace(/%([^%]+)%/g, function (_,n) { return process.env[n] });
+            var optionsCached = {
+                "silent": true,
+                "local": true
+            };
+
+            gpii.windows.rm(cachedFilesPath, optionsCached);
```

Comment:
How about just calling `rimraf.sync` directly here.

And, just remove the entire rm function above (make it empty).

Reply:
Doing that will leave us without the ability of using `gpii.windows.rm` from SR in other place (I know that right now isn't used in any other place but... don't feel like removing the ability is necessary), also, the error handling for `rimraf.sync` is already encapsulated in `gpii.windows.rm`. I don't know if it will have benefits to move that error handling here.